File: Domain/Entities/Item.cs
using Domain.Common;
using Domain.Enum;
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Item :BaseAuditableEntity
    {
        [Required]
        public string PartNumber { get; set; }
        [Required]
        public string SerialNumber { get; set; }
        [Required]
        public string OracleCode { get; set; }
        public string LocalCode { get; set; }
        public DateTime? OrderDate { get; set; }
        public int ExceededLimit { get; set; }
        public string AddVoucher { get; set; }

        public ItemStatus ItemStatus { get; set; } = ItemStatus.stored;
        public string Model { get; set; }
        public string Description { get; set; }
        public int BrandId { get; set; }
        public Brand Brand { get; set; }
        public int WarehouseId { get; set; }
        public Warehouse Warehouse { get; set; }
        public int? CustomerId { get; set; }
        public Customer? Customer { get; set; }
        public int? EngneerId { get; set; }
        public Engineer? Engineer { get; set; }
        public byte[]? Image { get; set; }
    }
}

File: API/client-app/src/app/shared/models/item.ts
import { IBrand } from "./brand"
import { ICustomer } from "./customer"
import { IEngineer } from "./engineer"
import { IWarehouse } from "./warehouse"

export interface IItemPagination {
  items: IItem[]
  pageNumber: number
  totalPages: number
  totalCount: number
  hasPreviousPage: boolean
  hasNextPage: boolean
}

export interface IDashboardItem {
  browwed: string
  sold: string
  store: string
  total: string
  degitalCheck: string
  brabouz: string
  panini: string
  xerox: string
  brands:any
  customers:any
  scanner: number[] 
  parts: number[] 
  printer: number[]
}


export interface IItem {
  partNumber: string
  serialNumber: string
  oracleCode: string
  localCode: string
  itemStatus: number
  addVoucher: number
  model: string
  description: string
  created: string
  createdBy: string
  orderDate: string
  itemType: string
  brand: string
  brandId: number
  warehouse: string
  customer: string
  engineer: string
  image: any
}

export interface IItemType {
  id: number
  name: string
  description: string
}

export interface IItemDetail {
  partNumber: string
  serialNumber: string
  oracleCode: string
  localCode: string
  itemStatus: number
  partStatus: number
  model: string
  description: string
  brand: IBrand
  id: number
  addVoucher: number
  warehouse: IWarehouse
  customer: ICustomer
  engineer: IEngineer
  orderDate: string
  created: string
  createdBy: string
  lastModified: string
  lastModifiedBy: string
  image: any

}
